exercises/diffie-hellman/canonical-data.json

```diff
@@ -0,0 +1,126 @@
+{ "exercise": "diffie-hellman"
+, "version" : "1.0.0"
+, "comments":
+    [ "                                                        "
+    , " There are irregularities in how this problem is being  "
+    , " tested across the current tracks:                      "
+    , "                                                        "
+    , " csharp, fsharp, elixer and go all consistently test    "
+    , " both for large and small prime numbers test cases.     "
+    , " They also assume that the application generates the    "
+    , " private key, not the user picks as specified by the    "
+    , " problem description.                                   "
+    , "                                                        "
+    , " ecmascript and javascript, on the other hand, only use "
+    , " the very small prime numbers as the test cases, and    "
+    , " ask the user to implement a private key generator.     "
+    , "                                                        "
+    , " The other major difference is that both javascript and "
+    , " ecmascript test boundaries and verify that their       "
+    , " implementations throw errors for specific cases.       "
+    , " The other implementations do no do this (perhaps       "
+    , " because they are all typed?).                          "
+    , "                                                        "
+    , " One last difference is that the go implementation uses "
+    , " hexadecimal notation for the bit Ints."
+    ]
+
+, "cases":
+    [ { "comments":
+          [ " In the following descriptions, p and g represent two "
+          , " prime numbers.                                       "
+          ]
+      , "description": "Private keys are generated in Range (1..p)"
+      , "property"   : "generatePrivateKey"
+      , "p": "23"
+      }
+    , { "description": "generatePrivateKey's output is random."
+      , "property"   : "generatePrivateKey"
+      }
+    , { "description": "Public key is correctly calculated from private key and two primes."
+      , "property"   : "getPublicKeyFromPrivateKey"
+      , "p": "23"
+      , "g": "5"
+      , "private_key": "6"
+      , "expected": "8"
+      }
+    , { "description": "Shared secret is correctly calculated."
+      , "property"   : "generateSharedSecret"
+      , "p": "23"
+      , "bob_public_key": "19"
+      , "alice_private_key": "6"
+      , "expected": "2"
+      }
+    , { "description": "Shared secret generated by Alice and Bob should be equal."
+      , "property"   : "generateSharedSecret"
+      }
+    , { "description": "Tests with very large primes."
```

Comment:
Are these tests necessary?
How are very large primes different from small primes?

Reply:
I personally don't think they are not needed.  But since so many of the current implementers (including golang) are doing this, I didn't want to presume.  Maybe they should go into a "optional" section?